### PR TITLE
fix: [CDS-40921]: form validation for infrastructure modal

### PR DESCRIPTION
**Pull request by AshwinBhatkal-Harness**
### Summary

Added form validation to infrastructure modal

#### Screenshots

Now:

https://user-images.githubusercontent.com/96711906/184805577-0f87ba38-235f-460f-872b-7736b4d1801f.mov

Before: Not available there exists BE validation now

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10517)

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/__tests__/InfrastructureModal.test.tsx [hunk 1]
```diff
@@ -42,11 +72,100 @@ describe('Infrastructure Modal Test', () => {
...
+
+    const kubernetesDeploymentType = screen.getAllByRole('checkbox')[0]
+    userEvent.click(kubernetesDeploymentType)
+
+    await waitFor(() => {
+      expect(screen.getByText('pipelineSteps.deploy.infrastructure.directConnection')).toBeInTheDocument()
+      expect(screen.getByText('pipelineSteps.deploymentTypes.kubernetes')).toBeInTheDocument()
+      expect(screen.getByText('pipelineSteps.deploy.infrastructure.viaCloudProvider')).toBeInTheDocument()
+      expect(screen.getByText('pipelineSteps.deploymentTypes.gk8engine')).toBeInTheDocument()
+    })
```
Comment (udayvunnam) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/__tests__/InfrastructureModal.test.tsx:104]: do we need to waitFor all of these assertions?

File: src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStage.module.scss [hunk 2]
```diff
@@ -5,18 +5,21 @@
  * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
  */
 
+.tabHeading {
+  font-size: 16px;
+  color: var(--black);
+  font-weight: 600;
+  line-height: 24px;
```
Comment (udayvunnam) [inline src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStage.module.scss:12]: nit: can make use of vars with this refactoring

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/__tests__/InfrastructureModal.test.tsx (hunk 1 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/__tests__/InfrastructureModal.test.tsx:104]: Yes. This is required to validate the headers in this scenario. Line 100 and 102 are group headers, while line 101 and 103 are group items

Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/__tests__/InfrastructureModal.test.tsx:104]: The second group is shown conditionally when FF is turned on

File: src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStage.module.scss (hunk 2 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStage.module.scss:12]: There is not font-size declared in ui core for 16px. I would have to use spacing and that would not be logically right

Ideally, this should been a part of the component itself but this class is being used across a lot of places, so did not go to make the change

Comment (udayvunnam) [reply src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStage.module.scss:12]: Hmm, should've been corrected at the start with designers. Seeing many such deviations with designs that we should take up with designers. Any that's not part of this PR. Good to go :)

**Review by udayvunnam — APPROVED** [review]

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/__tests__/InfrastructureModal.test.tsx (hunk 1 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/__tests__/InfrastructureModal.test.tsx]: Updated the test

**Review by udayvunnam — APPROVED** [review]

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureModal.tsx [hunk 3]
```diff
@@ -336,141 +355,154 @@ function BootstrapDeployInfraDefinition({
...
-                  }
+                    )
+                  }}
+                </Formik>
+              </Card>
+              {!infrastructureDefinition && (
+                <SelectDeploymentType
+                  viewContext="setup"
+                  selectedDeploymentType={selectedDeploymentType}
+                  isReadonly={false}
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureModal.tsx:418]: Why is this false? This should be based on permission

Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureModal.tsx:418]: Have marked it as false as there is no use case to have this disabled. One cannot view this after creation

**Review by bhavya-sinha — APPROVED** [review]

**Pull request merged**